src/delegates/NoteItemDelegate.h

```diff
@@ -60,14 +57,31 @@ class NoteItemDelegate: public QStyledItemDelegate
      */
     virtual void updateEditorGeometry(QWidget * editor, const QStyleOptionViewItem & option, const QModelIndex & index) const Q_DECL_OVERRIDE;
 
+public Q_SLOTS:
+    /**
+     * The slot which can watch for external changes in thumbnail display state.
+     * @param showThumbnailsForAllNotes Global flag for all notes.
+     * @param hideThumbnailsLocalUids  Map with local uids where the thumbails was manually hidden.
+     */
+    void setShowNoteThumbnailsState(bool showThumbnailsForAllNotes, const QSet<QString> & hideThumbnailsLocalUids);
+
+
 Q_SIGNALS:
     void notifyError(ErrorString error) const;  // clazy:exclude=const-signal-or-slot
 
 private:
     QString timestampToString(const qint64 timestamp, const qint64 timePassed) const;
 
 private:
-    bool                    m_showNoteThumbnails;
+    /**
+     * Current value of "shown thumbnails for all notes".
+     */
+    bool                m_showThumbnailsForAllNotes;
+    /**
+     * Set with local uids of notes where thumbnail was manually hidden.
+     */
+    QSet<QString>       m_hideThumbnailsLocalUids;
```

Comment:
Please move the names of these two data members one tab ahead so that they are in line vertically with other data members.